### PR TITLE
Remove cross-chain requests

### PR DESCRIPTION
**Pull request by darioush**
## Why this should be merged
No use cases depend on this

## How this works
Deletes code.
Note this bumps the RPC chain version and also sets the Minor version in anticipation of the new release (needed for the compatibility test).

## How this was tested
CI

### CODE REVIEW

**Review by joshua-kim — APPROVED** [review]

File: snow/engine/common/engine.go [hunk 1]
```diff
@@ -415,64 +415,8 @@ type AppGossipHandler interface {
...
-		chainID ids.ID,
-		requestID uint32,
-		appErr *AppError,
-	) error
-}
-
 type AppHandler interface {
 	NetworkAppHandler
-	CrossChainAppHandler
 }
```
Comment (StephenButtolph) [inline snow/engine/common/engine.go:420]: Should we remove `NetworkAppHandler` and just have `AppHandler`? Seems weird to have two identical interfaces.

File: snow/engine/common/sender.go [hunk 2]
```diff
@@ -192,36 +192,8 @@ type NetworkAppSender interface {
...
-	// [requestID].
-	SendCrossChainAppResponse(ctx context.Context, chainID ids.ID, requestID uint32, appResponseBytes []byte) error
-	// SendCrossChainAppError sends an application-level error to a CrossChainAppRequest
-	SendCrossChainAppError(ctx context.Context, chainID ids.ID, requestID uint32, errorCode int32, errorMessage string) error
-}
-
 // AppSender sends application (VM) level messages.
 // See also common.AppHandler.
 type AppSender interface {
 	NetworkAppSender
```
Comment (StephenButtolph) [inline snow/engine/common/sender.go:198]: Same comment about having duplicate interfaces

File: version/constants.go [hunk 3]
```diff
@@ -8,25 +8,22 @@ import (
...
+	RPCChainVMProtocol uint = 37
 )
 
 // These are globals that describe network upgrades and node versions
 var (
 	Current = &Semantic{
 		Major: 1,
 		Minor: 11,
-		Patch: 10,
+		Patch: 11,
```
[the remark this answers is not in the document]
Comment (darioush) [reply version/constants.go]: reverted

File: snow/engine/common/engine.go (hunk 1 above)
Comment (darioush) [reply snow/engine/common/engine.go:420]: inlined

File: snow/engine/common/sender.go (hunk 2 above)
Comment (darioush) [reply snow/engine/common/sender.go:198]: inlined

**Review by StephenButtolph — APPROVED** [review]

**Pull request merged**